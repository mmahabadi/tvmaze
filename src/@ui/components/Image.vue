<script setup lang="ts">
import { defineProps } from 'vue'

defineProps<{ src?: string; alt?: string }>()
</script>
<template>
  <img v-if="src" :src="src" :alt="alt" class="w-full h-full object-cover" />
  <img
    v-else
    class="w-full h-full object-cover"
    src="@/assets/image-placeholder.jpg"
    alt="image-placeholder"
  />
</template>
