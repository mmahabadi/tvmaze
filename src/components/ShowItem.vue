<script setup lang="ts">
import { RouterLink } from 'vue-router'
import { type Movie } from '@/types'
import IconPlay from '@/@ui/components/icons/IconPlay.vue'
import Image from '@/@ui/components/Image.vue'

type propType = Pick<Movie, 'name' | 'id'> & {
  image?: string
}

defineProps<propType>()
</script>
<template>
  <div class="show-item">
    <RouterLink :to="`/details/${id}`">
      <div>
        <Image :src="image" :alt="name" />
      </div>
      <div class="caption">
        <h3>{{ name }}</h3>
        <button>
          <IconPlay />
        </button>
      </div>
    </RouterLink>
  </div>
</template>

<style scoped>
.show-item {
  @apply overflow-hidden border border-gray-600 rounded-md p-1 transform duration-150;
  background-color: var(--bg-color);

  .caption {
    @apply absolute bottom-0 left-0 right-0 py-3 px-4 bg-black bg-opacity-70 text-white flex justify-between items-center;

    h3 {
      @apply overflow-hidden truncate font-semibold text-nowrap;
    }

    button {
      @apply flex justify-center items-center text-xs text-white transform duration-150 bg-white bg-opacity-20 rounded-md w-8 h-8;
      border: 2px solid var(--primary-color);
    }
  }

  &:hover {
    @apply relative scale-110 z-10;

    button {
      background-color: var(--primary-color);
    }
  }
  a {
    @apply w-full;
  }

  img {
    @apply w-full h-64 object-cover;
  }
}
</style>
