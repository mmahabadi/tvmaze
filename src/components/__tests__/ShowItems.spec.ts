import { describe, it, expect, beforeEach } from 'vitest'
import { mount, RouterLinkStub, VueWrapper } from '@vue/test-utils'
import ShowItem from '../ShowItem.vue'
import IconPlay from '@/components/icons/IconPlay.vue'
import Image from '@/@ui/components/Image.vue'

describe('ShowItem.vue', () => {
  let wrapper: VueWrapper<unknown>
  const props = {
    id: 1,
    name: 'Inception',
    image: 'inception.jpg'
  }

  beforeEach(() => {
    wrapper = mount(ShowItem, {
      props,
      global: {
        stubs: {
          RouterLink: RouterLinkStub,
          IconPlay,
          Image
        }
      }
    })
  })

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('renders details', () => {
    expect(wrapper.find('h3').text()).toBe('Inception')
    expect(wrapper.findComponent(Image).props('src')).toBe(props.image)
    expect(wrapper.findComponent(Image).props('alt')).toBe(props.name)
    expect(wrapper.findComponent(IconPlay).exists()).toBe(true)
  })

  it('renders the RouterLink with the correct URL', () => {
    const routerLink = wrapper.findComponent(RouterLinkStub)
    expect(routerLink.props().to).toBe(`/details/${props.id}`)
  })
})
