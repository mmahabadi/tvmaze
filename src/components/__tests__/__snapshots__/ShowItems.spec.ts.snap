// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ShowItem.vue > renders correctly 1`] = `
"<div data-v-a85b8742="" class="show-item"><a data-v-a85b8742="">
    <div data-v-a85b8742=""><img data-v-a85b8742="" src="inception.jpg" alt="Inception" class="w-full h-full object-cover"></div>
    <div data-v-a85b8742="" class="caption">
      <h3 data-v-a85b8742="">Inception</h3><button data-v-a85b8742=""><svg data-v-a85b8742="" class="svg-icon" style="width: 1em; height: 1em; vertical-align: middle; fill: currentColor; overflow: hidden;" viewBox="0 0 1024 1024" version="1.1" xmlns="http://www.w3.org/2000/svg">
          <path d="M852.727563 392.447107C956.997809 458.473635 956.941389 565.559517 852.727563 631.55032L281.888889 993.019655C177.618644 1059.046186 93.090909 1016.054114 93.090909 897.137364L93.090909 126.860063C93.090909 7.879206 177.675064-35.013033 281.888889 30.977769L852.727563 392.447107 852.727563 392.447107Z"></path>
        </svg></button>
    </div>
  </a></div>"
`;
